docs/source/start/declarative.rst

```diff
@@ -184,6 +187,44 @@ e.g. numeric properties. This example changes the ``min_card`` property of an
        min_card: 0
        max_card: .inf
 
+Extending container attributes of objects
+-----------------------------------------
+
+Moving possibly multiple objects that are in relation to the parent can
+be done with the ``extend:`` key. The following example would move the
+logical ``root_function`` into the ``.functions`` of the logical
+``root_component`` (parent):
+
+.. code-block:: yaml
+   :emphasize-lines: 2
+
+   - parent: !uuid 0d2edb8f-fa34-4e73-89ec-fb9a63001440
+     extend:
+       functions:
+         - !uuid f28ec0f8-f3b3-43a0-8af7-79f194b29a2d
+
+It is mandatory that the attribute that is to be extended has a
+one-to-many relationship to the parent which results into a
+:py:class:`~capellambse.model.common.accessors.ElementListCouplingMixin`.
+It is also possible to move not-yet created objects, but extra caution
+for declaring ``promise_id`` \s for resolving these promises
+successfully:
+
+.. code-block:: yaml
+   :emphasize-lines: 5,9
+
+   - parent: !uuid f28ec0f8-f3b3-43a0-8af7-79f194b29a2d
+     create:
+       functions:
+         - name: The promised one
+           promise_id: promised-fnc
+   - parent: !uuid 0d2edb8f-fa34-4e73-89ec-fb9a63001440
+       extend:
+         functions:
+           - !promise promised-fnc
```

Comment:
This bit of functionality (i.e. the combination `extend:`/move + `!promise`) doesn't really make sense to me. Why would you want to create the function somewhere, and then later *in the same file* move it somewhere else - instead of just creating it there in the first place? This can (and probably will at some point) create an issue in more complex scenarios with multiple references, where the actual order of modifications in the model isn't reliably known beforehand. (Which isn't the point of declarative modelling to begin with.)

Trying to catch this case while applying a YAML would probably introduce a fair bit of complexity, but the least we can do here is to not encourage it by making it one of our documented examples.

Reply:
You know more than me about the intricacies of the declarative-modelling implementation. But creating new references to promised (i.e. elements to be created) elements should work, no?